src/Nethermind/Nethermind.Db/DbExtensions.cs

```diff
@@ -147,57 +167,35 @@ public static void Delete(this IDb db, long key)
             db.Remove(key.ToBigEndianByteArrayWithoutLeadingZeros());
         }
 
-        public static TItem? Get<TItem>(this IDb db, Keccak key, IRlpStreamDecoder<TItem> decoder, LruCache<ValueKeccak, TItem> cache = null, bool shouldCache = true) where TItem : class
+        public static TItem? Get<TItem>(this IDb db, long blockNumber, ValueKeccak hash, IRlpStreamDecoder<TItem> decoder,
+            LruCache<ValueKeccak, TItem> cache = null, bool shouldCache = true) where TItem : class
         {
-            TItem item = cache?.Get(key);
-            if (item is null)
-            {
-                if (db is IDbWithSpan spanDb && decoder is IRlpValueDecoder<TItem> valueDecoder)
-                {
-                    Span<byte> data = spanDb.GetSpan(key);
-                    if (data.IsNull())
-                    {
-                        return null;
-                    }
-
-                    try
-                    {
-                        if (data.Length == 0)
-                        {
-                            return null;
-                        }
-
-                        var rlpValueContext = data.AsRlpValueContext();
-                        item = valueDecoder.Decode(ref rlpValueContext, RlpBehaviors.AllowExtraBytes);
-                    }
-                    finally
-                    {
-                        spanDb.DangerousReleaseMemory(data);
-                    }
-                }
-                else
-                {
-                    byte[]? data = db.Get(key);
-                    if (data is null)
-                    {
-                        return null;
-                    }
-
-                    item = decoder.Decode(data.AsRlpStream(), RlpBehaviors.AllowExtraBytes);
-                }
-            }
-
-            if (shouldCache && cache is not null && item is not null)
-            {
-                cache.Set(key, item);
-            }
+            Span<byte> dbKey = stackalloc byte[40];
```

Comment:
Can get away with a skiplocalsinit on this as is fully written in GetBlockNumPrefixedKey

Reply:
Done